newIDE/app/src/MainFrame/EditorContainers/HomePage/TeamSection/ManageEducationAccountDialog/ManageStudentRow.js

```diff
@@ -0,0 +1,204 @@
+    newPassword: string,
+  |}) => Promise<void>,
+|};
+
+const ManageStudentRow = ({
+  member,
+  isSelected,
+  isArchived,
+  onSelect,
+  onChangePassword,
+}: Props) => {
+  const { isMobile } = useResponsiveWindowSize();
+  const [isEditingPassword, setIsEditingPassword] = React.useState<boolean>(
+    false
+  );
+  const [
+    passwordEditionError,
+    setPasswordEditionError,
+  ] = React.useState<React.Node>(null);
+
+  const onEditPassword = React.useCallback(
+    async (newPassword: string) => {
+      if (member.password && newPassword === member.password) {
+        setIsEditingPassword(false);
+        return;
+      }
+      setPasswordEditionError(null);
+      if (newPassword.length < 8 || newPassword.length > 30) {
+        setPasswordEditionError(
+          <Trans>Your password must be between 8 and 30 characters long.</Trans>
+        );
+        return;
+      }
+      await onChangePassword({ userId: member.id, newPassword });
+      setIsEditingPassword(false);
+    },
+    [member.password, member.id, onChangePassword]
+  );
+
+  const usernameCell = (
+    <LineStackLayout alignItems="center" noMargin>
+      <Checkbox
+        style={{ margin: 0 }}
+        checked={isSelected}
+        onCheck={(e, checked) => {
+          onSelect(checked);
+        }}
+        uncheckedIcon={<CheckboxUnchecked />}
+        checkedIcon={<CheckboxChecked />}
+      />
+      {member.username ? (
+        <Text
+          allowSelection
+          style={{
+            ...textEllipsisStyle,
+            opacity: isArchived ? 0.6 : undefined,
+          }}
+        >
+          <b>{member.username}</b>
+        </Text>
+      ) : (
+        <Text
+          style={{
+            ...textEllipsisStyle,
+            opacity: isArchived ? 0.6 : 0.7,
+            // For italic characters not to be cut.
+            paddingRight: 1,
+          }}
+        >
+          <i>
+            <Trans>Not set</Trans>
+          </i>
+        </Text>
+      )}
+    </LineStackLayout>
+  );
+
+  const emailCell = (
+    <Text
+      style={{
+        ...textEllipsisStyle,
+        opacity: 0.7,
+      }}
+      allowSelection
+      noMargin
+    >
+      {member.email}
+    </Text>
+  );
+
+  const passwordCell = isEditingPassword ? (
+    <Line>
+      <AsyncSemiControlledTextField
+        margin="none"
+        autoFocus="desktop"
+        value={member.password || ''}
+        callback={onEditPassword}
+        errorText={passwordEditionError}
+        callbackErrorText={
+          <Trans>
+            An error occurred while changing the password. Please try again
+            later.
+          </Trans>
+        }
+        emptyErrorText={<Trans>Password cannot be empty</Trans>}
+        onCancel={() => setIsEditingPassword(false)}
+      />
+    </Line>
+  ) : (
+    <LineStackLayout alignItems="center">
+      <Text
+        noMargin
+        style={{
+          ...textEllipsisStyle,
+          opacity: isArchived ? 0.55 : 0.7,
```

Comment:
There is a lot of opacity: 0.7 for Texts... should we have a new color for it? Isn't "secondary" already working?

Reply:
Nit: Maybe it's just a matter of exporting a global constant somewhere that is 0.7 or 0.6, so that there is a bit of consistency in the code rather than magic numbers?

Reply:
No, it's not working but I will use variables with meaningful names to understand why it's as is